### PR TITLE
Add tests for character-count script

### PR DESCRIPTION
**Pull request by alex-ju**
This PR contains a basic set of tests for the character count component that were mistakenly missed from the [initial PR](https://github.com/alphagov/govuk-frontend/pull/959) by author's fault.

Issue: #1039

### CODE REVIEW

File: src/components/character-count/character-count.test.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+      expect(message).toEqual('You have 9 characters remaining')
+    })
+
+    it('counts up while typing and the limit is exceeded', async () => {
+      await page.goto(baseUrl + '/components/character-count/preview', { waitUntil: 'load' })
+
+      // Press 11 characters
+      await page.focus('.js-character-count')
+      for (let i = 0; i < 11; i++) {
```
Comment (36degrees) [inline src/components/character-count/character-count.test.js]: Not blocking: would `page.keyboard.type('A'.repeat(11))` have the same effect?

File: src/components/character-count/character-count.test.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  await page.close()
+})
+
+describe('Character count', () => {
+  describe('when JavaScript is unavailable or fails', () => {
+    it('shows the static message', async () => {
+      await page.setJavaScriptEnabled(false)
+
+      await page.goto(baseUrl + '/components/character-count/preview', { waitUntil: 'load' })
+      const message = await page.evaluate(() => document.body.querySelector('.govuk-character-count__message').innerHTML.trim())
```
Comment (36degrees) [inline src/components/character-count/character-count.test.js]: Non-blocking: Lines like this are very long, can we break them up?

```suggestion
      const message = await page.evaluate(() => {
        return document.body.querySelector('.govuk-character-count__message').innerHTML.trim()
      })
```

Alternatively, it looks like we might be able to use `page.$eval`? (not tested)

```suggestion
      const message = await page.$eval('.govuk-character-count__message', el => el.innerHTML.trim())
```

File: src/components/character-count/character-count.test.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  browser = global.__BROWSER__
+  page = await browser.newPage()
+})
+
+afterEach(async () => {
+  await page.close()
+})
+
+describe('Character count', () => {
+  describe('when JavaScript is unavailable or fails', () => {
```
Comment (36degrees) [inline src/components/character-count/character-count.test.js]: 👌 Great to see tests for this.

File: src/components/character-count/character-count.test.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      // Press 11 characters
+      await page.focus('.js-character-count')
+      for (let i = 0; i < 11; i++) {
+        await page.keyboard.press('A')
+      }
+
+      const textareaClasses = await page.evaluate(() => document.body.querySelector('.js-character-count').className)
+      const messageClasses = await page.evaluate(() => document.body.querySelector('.govuk-character-count__message').className)
+
+      expect(textareaClasses).toContain('govuk-textarea--error')
```
Comment (36degrees) [inline src/components/character-count/character-count.test.js]: Non-blocking: Would prefer one assertion per test.

**Comment by alex-ju** [comment]
@36degrees thanks. will update with your suggestions. `.repeat(11)` feels better 😅

File: src/components/character-count/character-count.test.js (hunk 2 above)
Comment (36degrees) [reply src/components/character-count/character-count.test.js]: Just tried `page.$eval` out in #1056 and it seems to work well for things like this 👍

File: src/components/character-count/character-count.test.js (hunk 1 above)
Comment (alex-ju) [reply src/components/character-count/character-count.test.js]: Repeat doesn't seem to be working, unfortunately, so I'll have to leave the form.

**Pull request merged**